### PR TITLE
Gamm multihop

### PR DESCRIPTION
**Pull request by Thunnini**
I've added the multi-hop functionality as well as some test code. However, I've only added one case for the multi-hop test that succeeds. Let's take a look initially and revise later as needed.

I also think it'd be good to add more test codes on a separate branch.

### CODE REVIEW

**Comment by Thunnini** [comment]
I've also removed the max spot price functionality as I added the multi-hop functionality for now.

Let's discuss whether this feature is necessary and if it is I can add it back in.

**Comment by Thunnini** [comment]
#33

File: x/gamm/keeper/grpc_query.go [hunk 1]
```diff
@@ -89,14 +89,104 @@ func (k Keeper) Records(ctx context.Context, req *types.QueryRecordsRequest) (*t
...
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if req.TokenInDenom == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid denom")
+	}
+
+	if req.TokenOutDenom == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid denom")
```
Comment (psaradev) [inline x/gamm/keeper/grpc_query.go:102]: `req.TokenInDenom` and `req.TokenOutDenom` have response.
A. Merge these two if statements
B. differentiate the response like "invalid out denom"

**Comment by psaradev** [comment]
A. Adding CLI tests would be great
B. Writing shell scripts on README to try it out in real mode is 👍.

**Pull request merged**